src/models/game/epics.js

```diff
@@ -0,0 +1,235 @@
+import {
+  startGame,
+  gameStarted,
+  fold,
+  gameFolded,
+  raise,
+  betRaised,
+  betNotRaised,
+  replace,
+  cardReplaced,
+  cardNotReplaced,
+  check,
+  gameWin,
+  gameLose,
+  gameTie,
+  nextPhase,
+  phaseAdvanced,
+  gameChecked,
+} from "./actions";
+import deckCreator from "lib/deckCreator";
+import handCheck from "lib/handCheck";
+
+const startGameEpic = (action$, state$) =>
+  action$.pipe(
+    ofType(startGame.type),
+    map(({ payload }) => {
+      const newDeck = deck(state$.value).slice();
+      const newPlayerHand = playerHand(state$.value).slice();
+      const newAiHand = aiHand(state$.value).slice();
+
+      const bet = 100;
+      const newPlayerBet = playerBet(state$.value);
+      const newAiBet = aiBet(state$.value);
+      const newPot = pot(state$.value);
+
+      const newAiMoney = aiMoney(state$.value);
+      const newPlayerMoney = playerMoney(state$.value);
+
+      const newPhase = 1;
+
+      newDeck.push(deckCreator());
+      const newDeckFlat = newDeck.flat();
+
+      const newerPlayerHand = newPlayerHand.concat(newDeckFlat);
+      const newerAiHand = newAiHand.concat(newDeckFlat);
+
+      newerPlayerHand.splice(0, 1);
+      newerPlayerHand.splice(5);
+      newerAiHand.splice(0, 6);
+      newerAiHand.splice(5);
+      newDeckFlat.splice(0, 10);
+
+      return gameStarted({
+        deck: newDeckFlat,
+        playerHand: newerPlayerHand,
+        aiHand: newerAiHand,
+        changedPlayerHand: Array(5).fill(null),
+        playerBet: newPlayerBet + bet,
+        aiBet: newAiBet + bet,
+        pot: newPot + bet * 2,
+        playerMoney: newPlayerMoney - bet,
+        aiMoney: newAiMoney - bet,
+        phase: newPhase,
+      });
+    })
+  );
+
+const foldEpic = (action$, state$) =>
+  action$.pipe(
+    ofType(fold.type),
+    map(({ payload }) => {
+      const newPlayerMoney = playerMoney(state$.value);
+      const newAiMoney = aiMoney(state$.value);
+      const newPhase = 4;
+      const newPot = pot(state$.value);
+
+      return gameFolded({
+        playerBet: 0,
+        aiBet: 0,
+        phase: newPhase,
+        pot: 0,
+        playerMoney: newPlayerMoney,
+        aiMoney: newAiMoney + newPot,
+      });
+    })
+  );
+
+const raiseEpic = (action$, state$) =>
+  action$.pipe(
+    ofType(raise.type),
+    map(({ payload }) => {
+      const newPlayerBet = playerBet(state$.value);
+      const newAiBet = aiBet(state$.value);
+      const bet = 100;
+
+      const newPlayerMoney = playerMoney(state$.value);
+      const newAiMoney = aiMoney(state$.value);
+      const newPot = pot(state$.value);
+
+      if (newPlayerMoney > 0) {
+        return betRaised({
+          playerMoney: newPlayerMoney - bet,
+          aiMoney: newAiMoney - bet,
+          pot: newPot + bet * 2,
+          playerBet: newPlayerBet + bet,
+          aiBet: newAiBet + bet,
+        });
+      } else if (newPlayerMoney <= 0) {
+        return betNotRaised({});
+      }
+    })
+  );
```

Comment:
you dont need bet not raused. You better use the condintion with a filter operator before map

Reply:
It is done.